xlators/storage/posix/src/posix-handle.c

```diff
@@ -844,14 +844,6 @@ posix_handle_soft(xlator_t *this, const char *real_path, loc_t *loc,
                    "symlink %s -> %s failed", oldpath, newstr);
             return -1;
         }
-
-        ret = sys_fstatat(dfd, newstr, &newbuf, AT_SYMLINK_NOFOLLOW);
-
-        if (ret) {
-            gf_msg(this->name, GF_LOG_WARNING, errno, P_MSG_HANDLE_CREATE,
-                   "stat on %s failed ", newstr);
-            return -1;
-        }
     }
 
     ret = sys_stat(real_path, &newbuf);
```

Comment:
Even i believe we don't need this because real_path stbuf is already saved in oldbuf.
Ideally We should compare real_path (ino/st_dev) with gfid path(ino/st_dev).

Reply:
You are right, this should be sys_fstatat(dfd, newstr, &newbuf, 0); i.e. follow symlink, ack if this is what you meant and I will re-send the patch

Reply:
Yes

Reply:
Done

Reply:
This is failing regression because of too many symlinks. I think it is just better to remove it. I am not sure if the stat on real_path is useful.